lib/tests/tests.c

```diff
@@ -3117,6 +3117,94 @@ compute_beta_coalescence_rate_fails(unsigned int num_ancestors, double alpha)
     return 0;
 }
 
+#define check_time_change(msp, t) \
+    do { \
+        const double tol = 1e-9; \
+        double g, g2, t2; \
+        g = (msp)->model.model_time_to_generations(&(msp)->model, t); \
+        t2 = (msp)->model.generations_to_model_time(&(msp)->model, g); \
+        CU_ASSERT_DOUBLE_EQUAL(t, t2, tol); \
+        g2 = (msp)->model.generation_rate_to_model_rate(&(msp)->model, t); \
+        CU_ASSERT_DOUBLE_EQUAL(g, g2, tol); \
+        g = (msp)->model.model_rate_to_generation_rate(&(msp)->model, t); \
+        t2 = (msp)->model.generation_rate_to_model_rate(&(msp)->model, g); \
+        CU_ASSERT_DOUBLE_EQUAL(t, t2, tol); \
+    } while(0)
+
+static void
+check_model_time_change_consistency(double population_size, double t)
+{
+    int ret;
+    int j, k;
+    msp_t msp;
+    const unsigned int n = 10;
+    const double psi = 0.5;
+    const double c = 5.0;
+    const double alphas[] = {1e-6, 0.1, 1, 1.5, 2};
+    const double truncation_points[] = {1e-6, 0.1, 1, 5};
+    sample_t *samples = malloc(n * sizeof(sample_t));
+    gsl_rng *rng = gsl_rng_alloc(gsl_rng_default);
+    recomb_map_t recomb_map;
+    tsk_table_collection_t tables;
+
+    ret = recomb_map_alloc_uniform(&recomb_map, 1.0, 1, true);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_init(&tables, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_FATAL(samples != NULL);
+    CU_ASSERT_FATAL(rng != NULL);
+    memset(samples, 0, n * sizeof(*samples));
+    ret = msp_alloc(&msp, n, samples, &recomb_map, &tables, rng);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+
+    ret = msp_set_simulation_model_hudson(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_smc(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_smc_prime(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_dtwf(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_wf_ped(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_dirac(&msp, population_size, psi, c);
```

Comment:
Be nice to do the same thing here with plausible values for psi and c

Reply:
Happy to do that if you can suggest some numbers.

Reply:
Let's say psi = {1e-6, 0.01, 0.5, 0.9, 1 -1e-6} and c = {0.01, 1, 5}